### PR TITLE
Add Typescript declaration file index.d.ts (#94)

### PR DESCRIPTION
**Pull request by krisdages**
Used the options documentation from the README.
Made educated guesses about the signatures for mapHeaders and mapValues

This PR contains:

- [ ] bugfix
- [ ] feature
- [ ] refactor
- [ ] tests
- [ ] documentation
- [X] metadata

### Breaking Changes?

- [ ] yes
- [X] no

If yes, please describe the breakage.

### Please Describe Your Changes

Added the index.d.ts declaration file for the interface described under API in the README.md
Should resolve issue #94

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,6 +30,7 @@
     "test": "ava test/test.js"
   },
   "dependencies": {
+    "@types/node": "^10.11.7",
```
Comment (shellscape) [inline package.json]: I'm not too keen on adding this as a dependency to the project. Is there another way to accomplish the appropriate typing without this dependency?

**Comment by Tuizi** [comment]
This TS definition is not working for me, I had to change it for:

```typescript
declare module 'csv-parser' {
  import { Transform } from 'stream';

  interface CsvParser extends Transform {}
  interface CsvParserOptions {
    escape?: string;
    headers?: ReadonlyArray<string> | boolean;
    mapHeaders?: (args: { header: string; index: number }) => string | null;
    mavValues?: (args: { header: string; index: number; value: any }) => any;
    newline?: string;
    quote?: string;
    raw?: boolean;
    separator?: string;
    skipLines?: number;
    maxRowBytes?: number;
    strict?: boolean;
  }

  const csvParser: (
    optionsOrHeaders?: CsvParserOptions | ReadonlyArray<string>
  ) => CsvParser;

  export = csvParser;
}
```

**Comment by shellscape** [comment]
@Tuizi would you be willing to submit your definition as a Pull Request? this one looks abandoned

**Comment by Tuizi** [comment]
Sure I will submit it @shellscape but you have to add `@types/node` as a dependency because of `Transform`. Maybe not the version 10.x, we should with which version of Node csv-parser works well.
Do you have an idea?

**Comment by shellscape** [comment]
@Tuizi cool. I don't think we need that dependency because `Transform` can be stubbed out. The one thing I want to avoid is shipping a dependency on a type lib in this module. If that can't be avoided, then adding the definition to https://github.com/DefinitelyTyped/DefinitelyTyped/ is preferable.

**Comment by shellscape** [comment]
Closing in favor of #104

**Pull request closed without merging**